### PR TITLE
Add object type as data  attr to notification

### PR DESCRIPTION
**Pull request by akhil1508**
- I use the `object-type` that used to be there in [NC 24 notifications markup](https://github.com/nextcloud/notifications/blob/3eb17d20b5c0d1d75754ab73257a7a7f05170266/src/App.vue#L29) to apply custom styles to notifications from certain apps
- It is removed NC 25 onwards :(
- The semantically correct way would be to make it a `data-object-type` attribute
- Please consider re-adding it through this PR as it breaks some styles that I want applied

### CODE REVIEW

**Comment by nickvergessen** [comment]
It is/was not removed, it should just be there in both ways now.
But I'm not really sure what you are trying to achieve. Can you link/share your code where you style it and why?

**Comment by akhil1508** [comment]
> It is/was not removed, it should just be there in both ways now.
@nickvergessen Prior to NC 25, the notification entry(div) used to look like:

```html
<div data-v-7df7d6b8="" data-v-082f9351="" data-id="9181" data-timestamp="1680675908000" class="notification" notification_id="9181" object_type="previewgenerator" object_id="5.2.2">
```
After NC 25, it looks like:
```html
<li data-v-7fd45550="" data-v-03956494="" data-id="672" data-timestamp="1680636752000" class="notification">
```

- The `object_type` attr is missing after NC 25..

>  But I'm not really sure what you are trying to achieve. Can you link/share your code where you style it and why?

I am building am app with a simple occ command and a notifier to send notifications to all users via occ command(similar to the `notification:generate` command but to all users and with custom icons)

Since the `object_type` is the only way to identify that this notification is coming from my app and not  some other one, I style it for e.g. like:

```css
.notification[object_type="my-notifications-app"] {
  background-color: red;
}
```

But since it is gone in NC 25, my styles no longer apply :(

**Comment by nickvergessen** [comment]
Ah, yeah so that was never intended to be part of the HTML body (since it's invalid). Exactly why the change was done.
Not sure I like styling being applied this way, but I guess it makes sense then

**Comment by nickvergessen** [comment]
Let's also add `data-app` and then we can do this.

Can you run:
```
npm ci && npm run build
```
and also commit the resulting js/ changes? Then our CI will also be happy.

**Comment by akhil1508** [comment]
> Let's also add data-app and then we can do this.
Can you run:
npm ci && npm run build
and also commit the resulting js/ changes? Then our CI will also be happy.

@nickvergessen Done

**Comment by akhil1508** [comment]
@nickvergessen Adding the `data-object-type` attribute in the `Notification` component always keeps it empty. I think we need to add in `NotificationApp` view itself

File: src/NotificationsApp.vue [hunk 1]
```diff
@@ -49,6 +49,8 @@
 							:index="2016" />
 						<Notification v-for="(n, index) in notifications"
 							:key="n.notificationId"
+							:object-type="n.objectType"
+							:app="n.app"
```
Comment (nickvergessen) [inline src/NotificationsApp.vue:53]: This is already populated with the `v-bind="n"` below.

When you removed it, you can rebuild again.
Afterwards (if you are familiar with git) you could squash all the commits into 1 (so we don't have 8 commits to change back and forth 3 lines in the end)

Comment (akhil1508) [reply src/NotificationsApp.vue:53]: Makes sense, I fix, squash and force push

Comment (akhil1508) [reply src/NotificationsApp.vue]: Fixed @nickvergessen

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**